Automattic/newspack-newsletters
# fix: custom namespace for Guzzle

### All Submissions:






### Changes proposed in this Pull Request:

Implements [`php-scoper`](https://github.com/humbug/php-scoper) to add a custom prefix to Guzzle namespace.

Closes #539.

### How to test the changes in this Pull Request:

1. In the master branch, install the plugin [Constant Contact Forms](https://wordpress.org/plugins/constant-contact-forms/) (it uses an older version of Guzzle that causes conflict)
2. With Constant Contact provider configured, attempt to create a new newsletter
3. Observe the fatal error
4. Switch to this branch, run `composer install` and confirm newsletter create/edit/publish works
5. Also confirm that the release build works:
    1. Clear the `vendor/` and `vendor_prefixed/` directories (**do not remove the `vendor_prefixed/` entirely, leave it empty instead**)
    2. Run `composer install --no-dev --no-scripts && composer dump-autoload && npm run release:archive`
    3.  Install generated zip file in a new WordPress installation and confirm the plugin works with Constant Contact set as provider and with the plugin mentioned in the first step

### Other information:
